fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTestValidator.java

```diff
@@ -0,0 +1,241 @@
+                    // multiple orders of magnitude larger than the repartition count. (i.e. if the high watermark is
+                    // 500,000 it shouldn't matter if the actual document count is 499,990 or 499,995).
+                    // Avoiding this might also mean that you end up with excessive transactions if you are usually adding
+                    // to the most-recent partition, namely, it would keep rebalancing by moving 1 record at a time to
+                    // get it below the high watermark.
+
+                    // If there is only one partition, it's totally fine for it to be any value less than maxPerPartition
+                    if (i < partitionInfos.size() - 2) {
+                        assertThat(allCounts, partitionInfo.getCount(),
+                                Matchers.allOf(
+                                        greaterThanOrEqualTo(minPerPartition),
+                                        lessThanOrEqualTo(maxPerPartition)));
+                    }
+                    assertTrue(usedIds.add(partitionInfo.getId()), () -> "Duplicate id: " + partitionInfo);
+                    final Tuple fromTuple = Tuple.fromBytes(partitionInfo.getFrom().toByteArray());
+                    if (i > 0) {
+                        assertThat(fromTuple, greaterThan(lastToTuple));
+                    }
+                    lastToTuple = Tuple.fromBytes(partitionInfo.getTo().toByteArray());
+                    assertThat(fromTuple, lessThanOrEqualTo(lastToTuple));
+
+                    LOGGER.debug(KeyValueLogMessage.of("Visited partition",
+                            "group", group,
+                            "documentsSoFar", visitedCount,
+                            "documentsInGroup", records.size(),
+                            "partitionInfo.count", partitionInfo.getCount()));
+                    validateDocsInPartition(recordStore, index, partitionInfo.getId(), groupingKey,
+                            Set.copyOf(records.subList(visitedCount, visitedCount + partitionInfo.getCount())),
+                            universalSearch);
+                    visitedCount += partitionInfo.getCount();
+                }
+            }
+
+        }
+    }
+
+    private List<LucenePartitionInfoProto.LucenePartitionInfo> getPartitionMeta(Index index,
+                                                                                Tuple groupingKey) {
+        try (FDBRecordContext context = contextProvider.get()) {
+            final FDBRecordStore recordStore = schemaSetup.apply(context);
+            LuceneIndexMaintainer indexMaintainer = (LuceneIndexMaintainer) recordStore.getIndexMaintainer(index);
+            return indexMaintainer.getPartitioner().getAllPartitionMetaInfo(groupingKey).join();
+        }
+    }
+
+
+    private void validateDocsInPartition(final FDBRecordStore recordStore, Index index, int partitionId, Tuple groupingKey,
```

Comment:
There is still `LuceneIndexTest.validateDocsInPartition` which I think is very similar; I should see if that can call this.